x-pack/plugins/transform/server/routes/api/error_utils.ts

```diff
@@ -138,3 +146,28 @@ export function wrapEsError(err: any, statusCodeToMessageMap: Record<string, any
   const message = statusCodeToMessageMap[statusCode];
   return new Boom(message, { statusCode });
 }
+
+interface EsError {
+  type: string;
+  reason: string;
+  line?: number;
+  col?: number;
+}
+
+/**
+ * Returns an error message based on the root cause
+ */
+function extractErrorMessage({ type, reason, line, col }: EsError): string {
```

Comment:
I'm not sure this is guaranteed to match all errors raised from elasticsearch.
Also, the `root_cause` may not have the most relevant information for the error.
In ML we pass the whole es error through to the client and display it when the user clicks the more info button.

Reply:
We also have a `extractErrorMessage` function in the `public` code of transforms which we bring over from the ML plugin, maybe we can do the same on the server side?

Reply:
It's definitely worth improving error reporting, but in this PR I focused on preserving the initial error message shape.
I'll create an issue to provide more detailed errors in the transform UI.